web/packages/teleport/src/UnifiedResources/useResources.ts

```diff
@@ -56,10 +76,100 @@ export function useResources(ctx: Ctx): ResourcesState {
     fetchInitial();
   }, [clusterId, search]);
 
+  const getWindowsLoginOptions = ({ name, logins }: Desktop) =>
+    makeDesktopLoginOptions(clusterId, name, logins);
+
+  const startRemoteDesktopSession = (username: string, desktopName: string) => {
+    const url = cfg.getDesktopRoute({
+      clusterId,
+      desktopName,
+      username,
+    });
+
+    openNewTab(url);
+  };
+
+  function getNodeLoginOptions(serverId: string) {
```

Comment:
I think that `getNodeLoginOptions` doesn't have to belong here. It doesn't really depend on more data than is available to the resource card (or perhaps even action button itself). The only reason why we refer to `fetchedData` is to get exactly the node that is displayed by the card. Why don't we pass the resource itself all the way up to the action button and construct login options right there?

The same seems to apply to other functions, too. `startSshSession` doesn't rely on any list component state, neither does `getWindowsLoginOptions`... This all might be much simpler if we pushed this all the way down to the leaf. Note that since these operations are stateless, the action buttons will still be dumb components.